src/prefect/new_flow_engine.py

```diff
@@ -0,0 +1,304 @@
+import asyncio
+from contextlib import asynccontextmanager
+from dataclasses import dataclass
+from typing import (
+    Any,
+    Coroutine,
+    Dict,
+    Generic,
+    Iterable,
+    Literal,
+    Optional,
+    TypeVar,
+    cast,
+)
+
+import anyio
+from typing_extensions import ParamSpec
+
+from prefect import Flow, Task, get_client
+from prefect.client.orchestration import PrefectClient
+from prefect.client.schemas import FlowRun, TaskRun
+from prefect.client.schemas.filters import FlowRunFilter
+from prefect.client.schemas.sorting import FlowRunSort
+from prefect.context import FlowRunContext
+from prefect.futures import PrefectFuture, resolve_futures_to_states
+from prefect.logging.loggers import flow_run_logger
+from prefect.results import ResultFactory
+from prefect.states import (
+    Pending,
+    Running,
+    State,
+    exception_to_failed_state,
+    return_value_to_state,
+)
+from prefect.utilities.asyncutils import A, Async
+from prefect.utilities.engine import (
+    _dynamic_key_for_task_run,
+    _resolve_custom_flow_run_name,
+    collect_task_run_inputs,
+    propose_state,
+)
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+@dataclass
+class FlowRunEngine(Generic[P, R]):
+    flow: Flow[P, Coroutine[Any, Any, R]]
+    parameters: Optional[Dict[str, Any]] = None
+    flow_run: Optional[FlowRun] = None
+    _is_started: bool = False
+    _client: Optional[PrefectClient] = None
+    short_circuit: bool = False
+
+    def __post_init__(self):
+        if self.parameters is None:
+            self.parameters = {}
+
+    @property
+    def client(self) -> PrefectClient:
+        if not self._is_started or self._client is None:
+            raise RuntimeError("Engine has not started.")
+        return self._client
+
+    @property
+    def state(self) -> State:
+        return self.flow_run.state  # type: ignore
+
+    async def begin_run(self) -> State:
+        new_state = Running()
+        state = await self.set_state(new_state)
+        while state.is_pending():
+            await asyncio.sleep(1)
+            state = await self.set_state(new_state)
+        return state
+
+    async def set_subflow_state(self, state: State) -> State:
+        """This appears to not be necessary"""
+        pass
```

Comment:
do we want to keep this stub for now or nix it?

Reply:
I'm in favor of nixing any code we can.